tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc

```diff
@@ -694,16 +630,17 @@ bool TRTEngineOp::ExecuteTrtEngine(OpKernelContext* ctx,
   }
 
   const bool kRetry = true;
-  if (trt_context_idx >= 1) {
+  if (trt_context_idx >= engine_context->execution_context.size()) {
     LOG(ERROR) << "Requested engine context with index " << trt_context_idx
-               << ", but only 1 context is present.";
+               << ", but only " << engine_context->execution_context.size()
+               << "contexts are present.";
```

Comment:
Can we somehow combine this checking and engine_context->execution_context[trt_context_idx] into a member function of engine_context?